tofhir-server/src/main/scala/io/tofhir/server/model/ToFhirRejectionHandler.scala

```diff
@@ -0,0 +1,66 @@
+package io.tofhir.server.model
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.MalformedRequestContentRejection
+import akka.http.scaladsl.server.{MethodRejection, RejectionHandler, UnsupportedRequestContentTypeRejection, Rejection}
+
+object ToFhirRejectionHandler {
+
+  /**
+   * Custom rejection handler to send proper error message to front-end on rejections.
+   */
+  private val rejectionHandler: RejectionHandler =
+    RejectionHandler.newBuilder()
+
+      /**
+       * Handles the cases type of the request data is wrong. Ex: String instead of a terminology system model.
+       * supportedTypes is not worked properly in the tested cases, it only included 'application/json'
+       */
+      .handle {
+        case UnsupportedRequestContentTypeRejection(supportedTypes) =>
+          complete(StatusCodes.UnsupportedMediaType -> UnsupportedMediaType("Unsupported payload format", s"Server refuses to accept the request because the type of the request data is not in a supported format").toString)

```

Comment:
Do not stringify the error with toString. Let marshallers transform the error into json serializaiton.

Reply:
Now, the general format of the error responses is the same as that code. There is currently no marshaller to transform errors. A proper marshaller will be created and this stringifying logic will be changed after marshaller is created.